src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -98,28 +98,13 @@ public bool IsSelected
             }
         }
 
-        public bool IsInput
+        public bool? IsInputNode
```

Comment:
It's totally legal to have a field called the same thing on the view model as on the model. 

In the purest MVVM model, this property would manage all the property change notifications so the view could stay up to date, and it would set the value on the model. Alas, we do not live in a perfect world, and many of our model types have property change notification built into them.